snowfakery/__init__.py

```diff
@@ -10,3 +11,35 @@
 version_file = Path(__file__).parent / "version.txt"
 with version_file.open() as f:
     version = f.read().strip()
+
+
+def generate_data(
+    yaml_file: Union[Path, str],
+    option: List[Tuple[str, str]] = [],
+    dburl: str = None,
+    target_number: Optional[Tuple[int, str]] = None,
+    debug_internals: bool = False,
+    output_format: str = None,
+    output_file: Path = None,
+    output_folder: Path = None,
+    continuation_file: Path = None,
+):
+    from .cli import generate_cli
+
+    if target_number:
+        assert isinstance(target_number[0], int)
```

Comment:
Could maybe use the `@typechecked` decorator from typeguard to do runtime checks against the type annotations? :)